### DISCUSSION TITLE
install-xcode when build image for macos errored out

### DISCUSSION THREAD

**Discussion by zoezhangmattr** [actions/virtual-environments#10341] [Image configuration]
https://github.com/actions/runner-images/blob/main/images/macos/scripts/build/Install-Xcode.ps1
using this file to install xcode when build selfhosted macos runner
is it just me or anyone encouter same issue: cant easily complete running this script
i am not powershell expert (might be one of the reasons), but its not stable to run this script to install the xcode somehow, it errors out, i had to add a few try catch to make sure download is ok, then to proceed, i am not sure what is wrong here.

**Comment by erik-bershel** [comment]
The macOS code is not ready to run for creating a self-hosted VM. There are a lot of solutions there that are tied to our platform and environment.
In this particular case, the problem is that this script is trying to access a private storage account that is not accessible outside of our infrastructure.

**Reply by zoezhangmattr** [reply]
thank you!
